packages/dotnet-sdk/src/TeamsFx/Configuration/ConfigurationMethods.cs

```diff
@@ -37,7 +38,7 @@ public static class TeamsFxConfigurationMethods
             botAuthOption.ClientId = authOptionsValue.ClientId;
             botAuthOption.ClientSecret = authOptionsValue.ClientSecret;
             botAuthOption.OAuthAuthority = authOptionsValue.OAuthAuthority;
-            botAuthOption.TenantId = authOptionsValue.TenantId;
+            botAuthOption.TenantId = (new Regex(@"[^\/]+$")).Match(authOptionsValue.OAuthAuthority).Value;
```

Comment:
is it possible that authority ends with `/`?

Reply:
also add a comment for the regex to explain it

Reply:
Get tenant id through bot framework instead of extracting from authority.